src/input.cpp

```diff
@@ -510,6 +510,10 @@ void input_manager::init_keycode_mapping()
     add_keyboard_char_keycode_pair( KEY_END,       translate_marker_context( "key name", "END" ) );
     add_keyboard_char_keycode_pair( '\n',          translate_marker_context( "key name", "RETURN" ) );
 
+    for( char c = 'A'; c <= 'Z'; c++ ) {
+        add_keyboard_char_keycode_pair( c - 64, string_format( "CTRL+%s", c ) );
+    }
+
     // function keys, as defined by ncurses
     for( int i = F_KEY_NUM_BEG; i <= F_KEY_NUM_END; i++ ) {
         // not marked for translation here, but specially handled in get_keyname so
```

Comment:
The same should be done for control characters.

Reply:
I think this should be fixed; lmk if I messed something up